### PR TITLE
[bug] fix issues with sklearn conformance for preview Ridge for 2024.6

### PR DESCRIPTION
**Pull request by icfaust**
### Description 

Preview Ridge is currently guarded by a daal_check_version which does not activate the new implementation in public CI.  Public CI checks sklearn conformance for 1.0-1.5 for preview.  When the 2024.6 drop occurs, the new Ridge implementation will be used, and will fail for sklearn conformance for preview. This solves the errors observed for falling back to sklearn after running onedal due to inheritance issues in the daal4py version of Ridge for array-like or tuple alpha values, and some incorrect checking and default parameters for sklearn < 1.2. 

Fixes # - _issue number(s) if exists_

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [ ] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Review by DDJHB — APPROVED** [review]
thanks for covering this, great job here!
